core/src/main/java/com/github/mizool/core/rest/errorhandling/ErrorMessageDto.java

```diff
@@ -30,21 +30,45 @@ public ErrorMessageDto combineWith(ErrorMessageDto other)
     private SetMultimap<String, ErrorDto> combineErrors(ErrorMessageDto other)
     {
         SetMultimap<String, ErrorDto> combined = HashMultimap.create();
-        errors.forEach(combined::putAll);
-        other.getErrors()
-            .forEach(combined::putAll);
+        if (errors != null)
+        {
+            errors.forEach(combined::putAll);
+        }
+        if (other.getErrors() != null)
+        {
+            other.getErrors()
+                .forEach(combined::putAll);
+        }
         return Multimaps.unmodifiableSetMultimap(combined);
     }
 
     private Map<String, Object> combineGlobalParameters(ErrorMessageDto other)
     {
+        Map<String, Object> result;
+
         Map<String, Object> otherGlobalParameters = other.getGlobalParameters();
+        if (globalParameters == null && otherGlobalParameters == null)
+        {
+            result = null;
+        }
+        else if (globalParameters != null && otherGlobalParameters == null)
+        {
+            result = globalParameters;
+        }
+        else if (globalParameters == null && otherGlobalParameters != null)
+        {
+            result = otherGlobalParameters;
+        }
+        else
+        {
+            Set<String> combinedKeys = Sets.union(globalParameters.keySet(), otherGlobalParameters.keySet());
 
-        Set<String> combinedKeys = Sets.union(globalParameters.keySet(), otherGlobalParameters.keySet());
+            result = combinedKeys.stream()
+                .collect(ImmutableMap.toImmutableMap(key -> key,
+                    key -> resolveValueInOrder(key, otherGlobalParameters, globalParameters)));
```

Comment:
Please rename lambda arg `key` to `value`.

(Yes, this was already wrong before.)

Reply:
I don't get it. We are streaming over combined keys, so does the lambda arg name `key` not make perfect sense here?

Reply:
Yes it does. :D Both the key function and the value function of the map collector get the `key` as parameter.

Reply:
Agreed, I was confused.